src/common/config.h

```diff
@@ -259,37 +345,51 @@ struct md_config_t {
   mutable Mutex lock;
 
   friend class test_md_config_t;
+protected:
+  static const std::unordered_map<std::string, config_option> config_options;
 };
 
-typedef enum {
-	OPT_INT, OPT_LONGLONG, OPT_STR, OPT_DOUBLE, OPT_FLOAT, OPT_BOOL,
-	OPT_ADDR, OPT_U32, OPT_U64, OPT_UUID
-} opt_type_t;
+template<typename T>
+struct get_typed_value_visitor : public boost::static_visitor<T> {
+  template<typename U,
+    typename boost::enable_if<boost::is_same<T, U>, int>::type = 0>
+      T operator()(U & val) {
+	return std::move(val);
+      }
+  template<typename U,
+    typename boost::enable_if_c<!boost::is_same<T, U>::value, int>::type = 0>
+      T operator()(U &val) {
+	throw boost::bad_get();
```

Comment:
I would just "assert" here since it would include the full backtrace of the offending ```get_val```